### ISSUE TITLE
fluid.dygraph.save_persistables保存的参数只有一部分

### ISSUE THREAD

**Issue by Exception-star** [PaddlePaddle/Paddle#19116]
为使您的问题得到快速解决，在建立Issue前，请您先通过如下方式搜索是否有相似问题:【搜索issue关键字】【使用labels筛选】【官方文档】

建立issue时，为快速解决问题，请您根据使用情况给出如下信息：
- 标题：简洁、精准描述您的问题，例如“ssd 模型前置lstm报错  ”
- 版本、环境信息：
    1）PaddlePaddle版本：请提供PaddlePaddle版本号，例如1.1或CommitID
    2）CPU：请提供CPU型号，MKL/OpenBlas/MKLDNN/等数学库的使用情况
    3）GPU：请提供GPU型号，CUDA和CUDNN版本号
    4）系统环境：请说明系统类型、版本（例如Mac OS 10.14），Python版本
- 模型信息
    1）模型名称 2）使用数据集名称 3）使用算法名称 4）模型链接
- 复现信息：如为报错，请给出复现环境、复现步骤
- 问题描述：请详细描述您的问题，同步贴出报错信息、日志/代码关键片段

Thank you for contributing to PaddlePaddle.
Before submitting the issue, you could search issue in the github.Probably there was a similar issue submitted or resolved before.
If there is no solution,please make sure that this is a issue of models including the following details:
**System information**
-PaddlePaddle version （eg.1.1）or CommitID
-CPU: including CPUMKL/OpenBlas/MKLDNN version
-GPU: including CUDA/CUDNN version
-OS Platform (eg.Mac OS 10.14)
-Python version
-Name of Models&Dataset/details of operator
**To Reproduce**
Steps to reproduce the behavior
**Describe your current behavior**
**Code to reproduce the issue**
**Other info / logs**

**Comment by Exception-star** [comment]
import paddle.fluid.dygraph as dygraph
import paddle.fluid as fluid


class Conv2d(dygraph.Layer):
    def __init__(self,
                 name_scope,
                 num_filters,
                 filter_size,
                 stride=1,
                 padding=0,
                 dilation=1,
                 param_attr=None,
                 bias_attr=None,
                 use_cudnn=True,
                 act='relu',
                 dtype='float32'):
        super(Conv2d, self).__init__(name_scope)
        self._conv = dygraph.Conv2D(self.full_name(),
                                    num_filters,
                                    filter_size,
                                    stride=stride,
                                    padding=padding,
                                    dilation=dilation,
                                    param_attr=param_attr,
                                    bias_attr=bias_attr,
                                    use_cudnn=use_cudnn,
                                    act=act,
                                    dtype=dtype)

    def forward(self, inputs):
        return self._conv(inputs)


class MaxPool2d(dygraph.Layer):
    def __init__(self, name_scope, pool_size, pool_stride, pool_padding=0):
        super(MaxPool2d, self).__init__(name_scope)
        self._pool = dygraph.Pool2D(self.full_name(),
                                    pool_size=pool_size,
                                    pool_type="max",
                                    pool_stride=pool_stride,
                                    pool_padding=pool_padding)

    def forward(self, inputs):
        return self._pool(inputs)


class vggblock(dygraph.Layer):
    def __init__(self, name_scope, num_filters, filter_size, conv_nums):
        super(vggblock, self).__init__(name_scope)
        self.layers = []
        for i in range(conv_nums):
            self.layers.append(Conv2d(self.full_name(), num_filters, filter_size=filter_size, stride=1, padding=1))

    def forward(self, inputs):
        x = inputs
        for layer in self.layers:
            x = layer(x)
        return x


class vgg(dygraph.Layer):
    def __init__(self, name_scope):
        super(vgg, self).__init__(name_scope)
        self.block1 = vggblock(self.full_name(), 32, 3, 3)
        self.block2 = vggblock(self.full_name(), 64, 3, 3)
        self.pool = MaxPool2d(self.full_name(), 2, 2)
        self.fc = dygraph.FC(self.full_name(), size=10, act='softmax')

    def forward(self, inputs):
        x = self.block1(inputs)
        x = self.pool(x)
        x = self.block2(x)
        x = self.pool(x)
        x = self.fc(x)
        return x


if __name__ == '__main__':
    import numpy as np
    x = np.random.randn(1, 3, 224, 224).astype(dtype=np.float32)
    with dygraph.guard():
        model = vgg('VGG')
        x = dygraph.to_variable(x)
        y = model(x)
        print(y.shape)
        fluid.dygraph.save_persistables(model.state_dict(), "save_dir", None)


save_dir里面只有FC的参数，没有卷积核的参数？

**Comment by Exception-star** [comment]
import paddle.fluid.dygraph as dygraph
import paddle.fluid as fluid

class Conv2d(dygraph.Layer):
def init(self,
name_scope,
num_filters,
filter_size,
stride=1,
padding=0,
dilation=1,
param_attr=None,
bias_attr=None,
use_cudnn=True,
act='relu',
dtype='float32'):
super(Conv2d, self).init(name_scope)
self._conv = dygraph.Conv2D(self.full_name(),
num_filters,
filter_size,
stride=stride,
padding=padding,
dilation=dilation,
param_attr=param_attr,
bias_attr=bias_attr,
use_cudnn=use_cudnn,
act=act,
dtype=dtype)

def forward(self, inputs):
    return self._conv(inputs)
class MaxPool2d(dygraph.Layer):
def init(self, name_scope, pool_size, pool_stride, pool_padding=0):
super(MaxPool2d, self).init(name_scope)
self._pool = dygraph.Pool2D(self.full_name(),
pool_size=pool_size,
pool_type="max",
pool_stride=pool_stride,
pool_padding=pool_padding)

def forward(self, inputs):
    return self._pool(inputs)
class vggblock(dygraph.Layer):
def init(self, name_scope, num_filters, filter_size, conv_nums):
super(vggblock, self).init(name_scope)
self.layers = []
for i in range(conv_nums):
self.layers.append(Conv2d(self.full_name(), num_filters, filter_size=filter_size, stride=1, padding=1))

def forward(self, inputs):
    x = inputs
    for layer in self.layers:
        x = layer(x)
    return x
class vgg(dygraph.Layer):
def init(self, name_scope):
super(vgg, self).init(name_scope)
self.block1 = vggblock(self.full_name(), 32, 3, 3)
self.block2 = vggblock(self.full_name(), 64, 3, 3)
self.pool = MaxPool2d(self.full_name(), 2, 2)
self.fc = dygraph.FC(self.full_name(), size=10, act='softmax')

def forward(self, inputs):
    x = self.block1(inputs)
    x = self.pool(x)
    x = self.block2(x)
    x = self.pool(x)
    x = self.fc(x)
    return x
if name == 'main':
import numpy as np
x = np.random.randn(1, 3, 224, 224).astype(dtype=np.float32)
with dygraph.guard():
model = vgg('VGG')
x = dygraph.to_variable(x)
y = model(x)
print(y.shape)
fluid.dygraph.save_persistables(model.state_dict(), "save_dir", None)

**Comment by phlrain** [comment]
https://github.com/PaddlePaddle/models/blob/develop/dygraph/resnet/train.py#L195
多层的sublayers 实现请参考这个，应该没有调用add_sublayer
